renderlib/MathUtil.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+#include "glm.h"
+
+struct LinearSpace3f
+{
+  glm::vec3 vx;
+  glm::vec3 vy;
+  glm::vec3 vz;
+
+  LinearSpace3f()
+    : vx(1.0f, 0.0f, 0.0f)
+    , vy(0.0f, 1.0f, 0.0f)
+    , vz(0.0f, 0.0f, 1.0f)
+  {
+  }
+  LinearSpace3f(glm::vec3 vx, glm::vec3 vy, glm::vec3 vz)
+    : vx(vx)
+    , vy(vy)
+    , vz(vz)
+  {
+  }
+
+  LinearSpace3f inverse() const;
+
+private:
+  float determinant() const;
+  LinearSpace3f transpose() const;
+  LinearSpace3f adjoint() const;
+};
+
+struct AffineSpace3f
+{
+public:
+  struct LinearSpace3f l;
+  glm::vec3 p;
+
+  AffineSpace3f()
+    : l()
+    , p()
+  {
+  }
+  AffineSpace3f(struct LinearSpace3f l, glm::vec3 p)
+    : l(l)
+    , p(p)
+  {
+  }
+  AffineSpace3f inverse() const;
+};
+
+inline glm::vec3
+xfmVector(const struct LinearSpace3f& xfm, glm::vec3 p)
+{
+  return xfm.vx * p.x + xfm.vy * p.y + xfm.vz * p.z;
+}
```

Comment:
What does `xfm` mean here?

Reply:
Yes, just shorthand for transform.  It's a word that gets used so much it becomes second nature to abbreviate it.